### ISSUE TITLE
 xref-find-definitions error

### ISSUE THREAD

**Issue by VasilVasilev93** [NicolasPetton/xref-js2#13]
I'm using Emacs 25.1.1 and js2-mode along with xref-js2. I've set up like described in readme:
```
(require 'xref-js2)

(define-key js2-mode-map (kbd "M-.") nil)
(add-hook 'js2-mode-hook (lambda ()
  (add-hook 'xref-backend-functions #'xref-js2-xref-backend nil t)))
```

This is the error after using M-.
**Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  call-process(nil nil t nil "--js" "--noheading" "--nocolor" "--ignore-dir" "bower_components" "--ignore-dir" "node_modules" "--ignore-dir" "build" "--ignore-dir" "lib" "--ignore" "*.min.js" "\\bsetupCustomBindings\\b[\\s]*[:=][^=]|function[\\s]+\\bsetupCustomBindings\$
  apply(call-process nil nil t nil ("--js" "--noheading" "--nocolor" "--ignore-dir" "bower_components" "--ignore-dir" "node_modules" "--ignore-dir" "build" "--ignore-dir" "lib" "--ignore" "*.min.js" "\\bsetupCustomBindings\\b[\\s]*[:=][^=]|function[\\s]+\\bsetupCustomBi$
  process-file(nil nil t nil "--js" "--noheading" "--nocolor" "--ignore-dir" "bower_components" "--ignore-dir" "node_modules" "--ignore-dir" "build" "--ignore-dir" "lib" "--ignore" "*.min.js" "\\bsetupCustomBindings\\b[\\s]*[:=][^=]|function[\\s]+\\bsetupCustomBindings\$
  apply(process-file nil nil t nil ("--js" "--noheading" "--nocolor" "--ignore-dir" "bower_components" "--ignore-dir" "node_modules" "--ignore-dir" "build" "--ignore-dir" "lib" "--ignore" "*.min.js" "\\bsetupCustomBindings\\b[\\s]*[:=][^=]|function[\\s]+\\bsetupCustomBi$
  xref-js2--find-candidates("setupCustomBindings" "\\bsetupCustomBindings\\b[\\s]*[:=][^=]|function[\\s]+\\bsetupCustomBindings\\b|class[\\s]+\\bsetupCustomBindings\\b|(?<!new)[^.]setupCustomBindings[\\s]*\\(")
  xref-js2--find-definitions("setupCustomBindings")
  xref-js2--xref-find-definitions("setupCustomBindings")
  #[514 "\300^A!\207" [xref-js2--xref-find-definitions] 4 "\n\n(fn BACKEND SYMBOL)"](xref-js2 "setupCustomBindings")
  apply(#[514 "\300^A!\207" [xref-js2--xref-find-definitions] 4 "\n\n(fn BACKEND SYMBOL)"] xref-js2 "setupCustomBindings")
  xref-backend-definitions(xref-js2 "setupCustomBindings")
  xref--find-xrefs("setupCustomBindings" definitions "setupCustomBindings" nil)
  xref--find-definitions("setupCustomBindings" nil)
  xref-find-definitions("setupCustomBindings")
  funcall-interactively(xref-find-definitions "setupCustomBindings")
  call-interactively(xref-find-definitions nil nil)
  command-execute(xref-find-definitions)**

**Comment by ajpauwels** [comment]
I'm running into the same problem. Haven't had the time to trace the source yet, any luck? On Emacs 25.3.1, maybe an incompatibility with the dev version?

SOLVED: Don't be an idiot like me and not read the dependencies. Just install `the silver searcher` in your package manager and you should be good to go.

**Comment by VasilVasilev93** [comment]
@ajpauwels thank you for your time, but I have the silver searcher installed along with the other dependencies, and it's not the case

**Comment by NicolasPetton** [comment]
What does `(executable-find "ag")` give you?

**Comment by gdanov** [comment]
I had the same problem and solved it by installing ag ...

**Comment by jjrh** [comment]
Same issue however I do have ag installed.

` (wrong-number-of-arguments (lambda (k l) (nthcdr 1 (assoc k l))) 5)`

```
(executable-find "ag")
"/usr/bin/ag"
```

```
$ which ag
/usr/bin/ag

```
```
$ ag --version                      
ag version 2.1.0

Features:
  +jit +lzma +zlib

```

**Comment by fjsousa** [comment]
Same issue here.

This is my [config](https://github.com/fjsousa/emacs.d/blob/master/customizations/javascript.el):
```
(define-key js-mode-map (kbd "M-.") nil)

(add-hook 'js2-mode-hook (lambda ()
  (add-hook 'xref-backend-functions #'xref-js2-xref-backend nil t)))
```
which come from @NicolasPetton blog post and I've got ag installed
```
ELISP> (executable-find "ag")
"/usr/local/bin/ag"
ELISP> 
```
Also I'm on Emacs 26.1

**Comment by NicolasPetton** [comment]
Is it the same stacktrace as the original report? The original report showed that "ag" wasn't found.

**Comment by fjsousa** [comment]
No, actually I don't get an error, the issue is with the same function but it just returns saying that no references where found.

**Comment by NicolasPetton** [comment]
Flávio Sousa <notifications@github.com> writes:

> No, actually I don't get an error, the issue is with the same function
> but it just returns saying that no references where found.

Ok, then it's probably something quite different.

Could you send me simple steps to reproduce the issue?

**Comment by fjsousa** [comment]
@NicolasPetton actually found my issue. I was working under `lib`, which was was being ignored by xref-js2. https://github.com/NicolasPetton/xref-js2/blob/master/xref-js2.el#L56

**Comment by NicolasPetton** [comment]
Flávio Sousa <notifications@github.com> writes:

> @NicolasPetton actually found my issue. I was working under `lib`,
> which was was being ignored by
> xref-js2. https://github.com/NicolasPetton/xref-js2/blob/master/xref-js2.el#L56

Ok.  Maybe this should be documented on the readme :)
